src/futils.c

```diff
@@ -26,27 +26,25 @@ int git_futils_mkpath2file(const char *file_path, const mode_t mode)
 
 int git_futils_mktmp(git_str *path_out, const char *filename, mode_t mode)
 {
-	int fd;
-	mode_t mask;
-
-	p_umask(mask = p_umask(0));
+	const int open_flags = O_RDWR | O_CREAT | O_EXCL | O_BINARY | O_CLOEXEC;
+	int fd = -1;
+	unsigned int tries = TMP_MAX;
 
 	git_str_sets(path_out, filename);
 	git_str_puts(path_out, "_git2_XXXXXX");
 
 	if (git_str_oom(path_out))
 		return -1;
 
-	if ((fd = p_mkstemp(path_out->ptr)) < 0) {
-		git_error_set(GIT_ERROR_OS,
-			"failed to create temporary file '%s'", path_out->ptr);
-		return -1;
+	while (fd == -1 && tries--) {
+		if (mktemp(path_out->ptr) == NULL)
+			break;
```

Comment:
Using mktemp could be problematic on some systems.
From [www.man7.org](https://www.man7.org/linux/man-pages/man3/mktemp.3.html):
> Some implementations follow 4.3BSD and replace XXXXXX by the current process ID and a single letter, so that at most 26 different names can be returned.